### PR TITLE
Unit tests and fix for OnResetTimeout feature

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
This PR is **[ready]** for review.

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md

- Unit tests for OnResetTimeout feature
- Add methodName check and specific check for DialNumber 
- Add Request struct 

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/reset_timeout_handler_impl.cc [hunk 1]
```diff
@@ -67,20 +68,30 @@ void ResetTimeoutHandlerImpl::on_event(const event_engine::Event& event) {
   const auto event_id = event.id();
   if (hmi_apis::FunctionID::BasicCommunication_OnResetTimeout == event_id) {
     const smart_objects::SmartObject& message = event.smart_object();
+    auto method_name = MessageHelper::HMIFunctionIDFromString(
+        message[strings::msg_params][strings::method_name].asString());
+
+    if (hmi_apis::FunctionID::BasicCommunication_DialNumber == method_name ||
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/reset_timeout_handler_impl.cc]: @LitvinenkoIra whats wrong with dial number ? Please add comment

File: src/components/application_manager/src/message_helper/message_helper.cc [hunk 2]
```diff
@@ -288,6 +288,17 @@ hmi_apis::Common_Language::eType MessageHelper::CommonLanguageFromString(
   return hmi_apis::Common_Language::INVALID_ENUM;
 }
 
+hmi_apis::FunctionID::eType MessageHelper::HMIFunctionIDFromString(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/message_helper/message_helper.cc]: @LitvinenkoIra do we have Message Helper test to add test to this new function?

File: src/components/application_manager/test/reset_timeout_handler_test.cc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  event.set_smart_object(notification_msg);
+
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, default_timeout())
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kTimeout));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/reset_timeout_handler_test.cc]: @LitvinenkoIra why don't you check first 2 arguments of expectation ?

File: src/components/application_manager/test/reset_timeout_handler_test.cc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  event.set_smart_object(notification_msg);
+
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, default_timeout())
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kTimeout)).Times(0);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/reset_timeout_handler_test.cc]: @LitvinenkoIra right here we can miss the kTimeout value for expectation

File: src/components/include/application_manager/reset_timeout_handler.h [hunk 5]
```diff
@@ -33,8 +33,28 @@
 #ifndef SRC_COMPONENTS_INCLUDE_APPLICATION_MANAGER_RESET_TIMEOUT_HANDLER_H_
 #define SRC_COMPONENTS_INCLUDE_APPLICATION_MANAGER_RESET_TIMEOUT_HANDLER_H_
 
+#include <stdint.h>
+
 namespace application_manager {
 namespace request_controller {
+
+/*
+ * @brief Structure for mobile request
```
Comment (LuxoftAKutsan) [inline src/components/include/application_manager/reset_timeout_handler.h]: @LitvinenkoIra too pure description, another developer may be confused . Please specify for what purpose this struct will be used.

File: src/components/application_manager/src/reset_timeout_handler_impl.cc (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/src/reset_timeout_handler_impl.cc]: @LuxoftAKutsan comment added 54ad2b0276ca7b29a0173568af9d0c20a6cc229e

File: src/components/include/application_manager/reset_timeout_handler.h (hunk 5 above)
Comment (LitvinenkoIra) [reply src/components/include/application_manager/reset_timeout_handler.h]: @LuxoftAKutsan description changed 54ad2b0

File: src/components/application_manager/test/reset_timeout_handler_test.cc (hunk 4 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/test/reset_timeout_handler_test.cc]: @LuxoftAKutsan fixed 7d67c458fed62cd11e953954fa6285e19842d46c

File: src/components/application_manager/test/reset_timeout_handler_test.cc (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/test/reset_timeout_handler_test.cc]: @LuxoftAKutsan fixed 7d67c458fed62cd11e953954fa6285e19842d46c

File: src/components/application_manager/include/application_manager/message_helper.h [hunk 6]
```diff
@@ -786,6 +786,9 @@ class MessageHelper {
   static hmi_apis::Common_Language::eType CommonLanguageFromString(
       const std::string& language);
 
+  static hmi_apis::FunctionID::eType HMIFunctionIDFromString(
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/components/application_manager/include/application_manager/message_helper.h]: @LuxoftAKutsan done 2d523133b33badd56fa1c4b17332a3523d4fb905

File: src/components/application_manager/src/message_helper/message_helper.cc (hunk 2 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/src/message_helper/message_helper.cc]: @LuxoftAKutsan done 2d523133b33badd56fa1c4b17332a3523d4fb905

**Pull request merged**